common/src/main/java/io/netty/util/ThreadDeathWatcher.java

```diff
@@ -116,18 +121,28 @@ private static void schedule(Thread thread, Runnable task, boolean isWatch) {
      *
      * @return {@code true} if and only if the watcher thread has been terminated
      */
-    public static boolean awaitInactivity(long timeout, TimeUnit unit) throws InterruptedException {
-        if (unit == null) {
-            throw new NullPointerException("unit");
-        }
-
-        Thread watcherThread = ThreadDeathWatcher.watcherThread;
-        if (watcherThread != null) {
-            watcherThread.join(unit.toMillis(timeout));
-            return !watcherThread.isAlive();
-        } else {
-            return true;
+    public static boolean awaitInactivity(long amount, TimeUnit unit) throws InterruptedException {
```

Comment:
doesn't this have the same race condition described in https://github.com/netty/netty/pull/6104#discussion_r91187728?